File: solutions/csharp/2352-equal-row-and-column-pairs.cs
public class Solution {
    public int EqualPairs(int[][] grid) {
        // Get grid size
        int n = grid.Length;
        // Create transposed grid
        int[][] transposeGrid = new int[n][];
        for (int i = 0; i < n; i++) {
            transposeGrid[i] = new int[n];
        }
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                transposeGrid[j][i] = grid[i][j];
            }
        }
        // Initialize counter for equal row-column pairs
        int count = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                // Compare i-th row of grid with j-th row of transposed grid
                if (Enumerable.SequenceEqual(grid[i], transposeGrid[j])) {
                    // Increment count if pair is equal
                    count++;
                }
            }
        }
        // Return total count of equal pairs
        return count;
    }
}

File: solutions/python/2352-equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        count = 0 
        n = len(grid)
        row_counter = collections.Counter(tuple(row) for row in grid) 
        for c in range(n): 
            col = [grid[i][c] for i in range(n)] 
            count += row_counter[tuple(col)] 
        return count 
                                                                                                                
